src/pages/index.js

```diff
@@ -1,128 +1,247 @@
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getCardElement();
+}
+//Click Card Function
+function handleImageClick(link, name) {
+  popupWithImage.open({ name, link });
+}
+
+function handleSubmit(
+  request,
+  popupInstance,
+  reset,
+  loadingText = "Saving..."
+) {
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      popupInstance.close();
 
-function handleImageClick(name, link) {
-  cardImagePopup.open(name, link);
+      if (reset) {
+        popupInstance.reset();
+      }
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
 }
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  const cardElement = card.generateCard();
-  return cardElement;
+function handleEditProfileSubmit(inputValues) {
+  function makeRequest() {
+    return api.updateUserInfo(inputValues).then((userData) => {
+      userInfo.setUserInfo(userData);
+    });
+  }
+  handleSubmit(makeRequest, profileEditModal, true);
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  cardList.addItem(cardElement);
+function handleAddCardSubmit(inputValues) {
+  function makeRequest() {
+    return api.createCard(inputValues).then((res) => {
+      renderCard(res);
+    });
+  }
+  handleSubmit(makeRequest, addCardPopup, true);
 }
-/* -------------------------------------------------------------------------- */
-/*                               Event Handlers                               */
-/* -------------------------------------------------------------------------- */
-/* ------------------------------ Profile Edit ------------------------------ */
-function handleProfileEditSubmit(userData) {
-  profileInfo.setUserInfo(userData);
-  profileEditPopup.close();
+
+function handleDeleteButton(card) {
+  deleteImageModal.open();
+  return deleteImageModal.handleDelete(() => {
+    function makeRequest() {
+      return api.deleteCard(card.id).then(() => {
+        card.deleteCard();
+      });
+    }
+    handleSubmit(makeRequest, deleteImageModal, false, "Deleting...");
+  });
 }
-function handleCardAddSubmit(formValues) {
-  const name = formValues.title;
-  const link = formValues.link;
-  renderCard({ name, link });
-  cardAddPopup.close();
-  cardAddForm.reset();
-  formValidators[cardAddForm.getAttribute("id")].disableButton();
+
+function handleLikeButton(card) {
+  if (card.isLiked) {
+    api
+      .deleteLike(card.id)
+      .then(() => {
+        card.handleIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.handleIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors